### ISSUE TITLE
Can't create partitioned tables with pyspark DataFrameWriterV2 

### ISSUE THREAD

**Issue by alex-shchetkov** [apache/iceberg#3088]
Expectation: Using `writeTo().partitionedBy("x")` creates a partitioned table
Reality: `writeTo().partitionedBy("x")` creates a partition-less table

Quick Repro, using iceberg version 0.12:
```
df = spark.createDataFrame([{"col1": "two", "tenant_id": "123"}, 
                            {"col1": "four", "tenant_id": "456"}])
writer = (df
         .sortWithinPartitions("tenant_id")
         .writeTo(f"iceberg.ice_db.local_partition_test")
         .partitionedBy(
             "tenant_id"
         )
         .create()
)
```

The above creates a partitionless table:
```
"partition-spec" : [ ],
  "default-spec-id" : 0,
  "partition-specs" : [ {
    "spec-id" : 0,
    "fields" : [ ]
  } ],
```
------
Works fine when using sql directly:

```
df.createOrReplaceGlobalTempView("partition_test")
spark.sql(f"""
       create table iceberg.ice_db.local_partition_test2 using iceberg
       PARTITIONED BY (tenant_id)
       as select * from global_temp.partition_test order by tenant_id
       """).show()
```
Produces:
```
"partition-spec" : [ {
    "name" : "tenant_id",
    "transform" : "identity",
    "source-id" : 2,
    "field-id" : 1000
  } ],
```

**Comment by jeff303** [comment]
@alex-shchetkov , I think you need `.using("iceberg")` before `.create()` in the first snippet.

**Comment by peay** [comment]
@alex-shchetkov I had the same issue on Spark 3.1.2, and noticed it worked in Scala but not with `pyspark`.

I believe that the root cause is https://github.com/apache/spark/commit/33deeb35f1c994328b577970d4577e6d9288bfc2, fixed in `pyspark` 3.1.3 and other patches releases.
